ansible-playbooks/roles/deploy/tasks/main.yml

```diff
@@ -0,0 +1,178 @@
+---
+
+- name: Validate ENV variable
+  fail:
+    msg: "Required variable ENVhas not been provided"
+  when: (ENV is undefined) or (ENV| length == 0)
+
+- name: Validate none local deployment variables
+  fail:
+    msg: "Required variable {{ item }} has not been provided for {{ ENV }} deployment"
+  when:
+    - ENV != "local" 
+    - (vars[item] is undefined) or (vars[item]| length == 0)
+  loop:
+    - VAULT_ADDRESS
+    - VAULT_TOKEN
+  
+- name: Validate local deployment variables
+  fail:
+    msg: "Required variable {{ item }} has not been provided for local deployment"
+  when:
+    - ENV == "local" 
+    - (vars[item] is undefined) or (vars[item]| length == 0)  
+  loop:
+    - PULL_SECRET
```

Comment:
ansible will automatically fail when anything in {{ VAR }} is not defined. So, next time, we don't have to check this.